node-manager/src/nodemanager.rs

```diff
@@ -565,26 +580,52 @@ impl NodeManager {
     }
 
     #[wasm_bindgen]
-    pub async fn decode_invoice(&self, _invoice: String) -> Result<MutinyInvoice, MutinyJsError> {
-        todo!()
+    pub async fn decode_invoice(&self, invoice: String) -> Result<MutinyInvoice, MutinyJsError> {
+        Invoice::from_str(&invoice)
+            .map(|i| i.into())
+            .map_err(|e| e.into())
     }
 
     #[wasm_bindgen]
-    pub async fn get_invoice(&self, _invoice: String) -> Result<MutinyInvoice, MutinyJsError> {
-        todo!()
+    pub async fn get_invoice(&self, invoice: String) -> Result<MutinyInvoice, MutinyJsError> {
+        let nodes = self.nodes.lock().await;
+        let inv_opt: Option<MutinyInvoice> = nodes
+            .iter()
+            .find_map(|(_, n)| n.get_invoice(invoice.clone()).ok());
```

Comment:
We are going to be parsing `invoice` into an `Invoice` n times here. It would be better if `get_invoice` took in an `Invoice` type instead of `String`. Would be more efficient and makes that function more typesafe

Reply:
fixed, passing it through now